soc/xtensa/intel_adsp/cavs/include/intel_icl_adsp/adsp_memory.h

```diff
@@ -22,15 +22,6 @@
 #define RAM_BASE (L2_SRAM_BASE + CONFIG_HP_SRAM_RESERVE + VECTOR_TBL_SIZE)
 #define RAM_SIZE (L2_SRAM_SIZE - CONFIG_HP_SRAM_RESERVE - VECTOR_TBL_SIZE)
 
-/* Host shared memory windows */
-#define HP_SRAM_WIN0_BASE (L2_SRAM_BASE + CONFIG_ADSP_WIN0_OFFSET)
-#define HP_SRAM_WIN0_SIZE 0x2000
-#define HP_SRAM_WIN2_BASE (L2_SRAM_BASE + CONFIG_ADSP_WIN2_OFFSET)
-/* window 2 size is variable */
-#define HP_SRAM_WIN2_SIZE (CONFIG_ADSP_WIN3_OFFSET - CONFIG_ADSP_WIN2_OFFSET)
-#define HP_SRAM_WIN3_BASE (L2_SRAM_BASE + CONFIG_ADSP_WIN3_OFFSET)
-#define HP_SRAM_WIN3_SIZE 0x2000
-
```

Comment:
@nashif So in SOF side, these definitions are traditionally done (for each platform) in  src/platform/PLATFORM/include/platform/lib/memory.h
As we need to support non-Zephyr SOF builds as well in SOF, we've kept these definitions in place for platforms like "tgl". The definitions are part of the contract between SOF host and FW, so it's not like FW can change these values at its leisure, there are essentially part of the ABI contract.

Now for SOF MTL, this only supports a Zephyr build, so @aborisovich et al have done the right thing and used the Zephyr definitions:
sof/src/platform/meteorlake/include/platform/lib/memory.h

So if you now remove the old definitions in adsp_memory.h, you'll get a build error in SOF app. I think the SOF code could be updated to lookup the numbers from DTS directly (and this can be done for MTL as it's Zephyr only anyways). But transition needs to be managed to avoid build breakages inbetween.

I don't think this there is a big risk for duplicated definitions for these. If a the values are modified, you'll also break the host (i.e. this is picked up by the test sets very quickly).

Reply:
yes, this can worked around this for now.